src/compas/robots/model/robot.py

```diff
@@ -550,8 +565,78 @@ def get_base_link_name(self):
         >>> robot.get_base_link_name()
         'world'
         """
+        try:
+            for joint in self.iter_joint_chain():
+                if joint.is_configurable():
+                    return joint.parent.link
+        except:  # noqa: E722
+            pass
+
+        # If we cannot get the joint chain, assume ordered joints
+        # as a more or less decent default
         joints = self.get_configurable_joints()
-        return joints[0].parent.link
+        if joints and joints[0].parent:
+            return joints[0].parent.link
+
+        return None
+
+    @property
+    def rcf(self):
+        """Robot Coordinate Frame.
+
+        The RCF can be set either as a fixed joint in the model, or as an explicit frame
+        used to reposition the model. If both are used, the former takes precedence over the latter.
+
+        Returns
+        -------
+        :class:`compas.geometry.Frame`
+            Robot coordinate frame, usually placed at the base of the robot and attached to a ``world`` link.
+        """
+        return Frame.from_transformation(self._root_transformation)
+
+    @rcf.setter
+    def rcf(self, frame):
```

Comment:
There generally seems to be a lot of repeated code between the `rcf` setter and the `_root_transformation`.  Is there a nice way to compress these things?

Reply:
also done at https://github.com/compas-dev/compas/pull/762/commits/a4a3e8f86bfd3d99c4e14f781cfb39ccebd00bab (to some extent at least)